src/Sql/AbstractSql.php

```diff
@@ -1,59 +1,53 @@
 <?php
 /**
- * Zend Framework (http://framework.zend.com/)
- *
- * @link      http://github.com/zendframework/zf2 for the canonical source repository
- * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
- * @license   http://framework.zend.com/license/new-bsd New BSD License
+ * @see       https://github.com/zendframework/zend-db for the canonical source repository
+ * @copyright Copyright (c) 2005-2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-db/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Zend\Db\Sql;
 
 use Zend\Db\Adapter\Driver\DriverInterface;
 use Zend\Db\Adapter\ParameterContainer;
 use Zend\Db\Adapter\Platform\PlatformInterface;
-use Zend\Db\Sql\Platform\PlatformDecoratorInterface;
 use Zend\Db\Adapter\Platform\Sql92 as DefaultAdapterPlatform;
+use Zend\Db\Sql\Exception\InvalidArgumentException;
+use Zend\Db\Sql\Exception\RuntimeException;
+use Zend\Db\Sql\Platform\PlatformDecoratorInterface;
+use function count;
+use function gettype;
+use function implode;
+use function is_array;
+use function is_string;
+use function preg_replace;
+use function rtrim;
+use function sprintf;
+use function vsprintf;
 
 abstract class AbstractSql implements SqlInterface
 {
-    /**
-     * Specifications for Sql String generation
-     *
-     * @var string[]|array[]
-     */
+    /** @var string[]|array[] Specifications for Sql String generation */
     protected $specifications = [];
 
-    /**
-     * @var string
-     */
+    /** @var string[] */
```

Comment:
this is invalid type, values are mixed, keys are sting

Reply:
changed to mixed[]